src/backend/distributed/commands/role.c

```diff
@@ -895,6 +912,57 @@ GenerateGrantRoleStmtsOfRole(Oid roleid)
 }
 
 
+/*
+ * GenerateSecLabelOnRoleStmts generates the SecLabelStmts for the role
+ * whose oid is roleid.
+ */
+static List *
+GenerateSecLabelOnRoleStmts(Oid roleid, char *rolename)
+{
+	List *secLabelStmts = NIL;
+
+	/*
+	 * Note that roles are shared database objects, therefore their
+	 * security labels are stored in pg_shseclabel instead of pg_seclabel.
+	 */
+	Relation pg_shseclabel = table_open(SharedSecLabelRelationId, AccessShareLock);
+	ScanKeyData skey[1];
+	ScanKeyInit(&skey[0], Anum_pg_shseclabel_objoid, BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(roleid));
+	SysScanDesc scan = systable_beginscan(pg_shseclabel, SharedSecLabelObjectIndexId,
+										  true, NULL, 1, &skey[0]);
+
+	HeapTuple tuple = NULL;
+	while (HeapTupleIsValid(tuple = systable_getnext(scan)))
+	{
+		SecLabelStmt *secLabelStmt = makeNode(SecLabelStmt);
+		secLabelStmt->objtype = OBJECT_ROLE;
+		secLabelStmt->object = (Node *) makeString(pstrdup(rolename));
+
+		Datum datumArray[Natts_pg_shseclabel];
+		bool isNullArray[Natts_pg_shseclabel];
+
+		heap_deform_tuple(tuple, RelationGetDescr(pg_shseclabel), datumArray,
+						  isNullArray);
+
+		secLabelStmt->provider = TextDatumGetCString(
+			datumArray[Anum_pg_shseclabel_provider - 1]);
+		if (!isNullArray[Anum_pg_shseclabel_label - 1])
```

Comment:
While this can be provided to be NULL for a SecLabelStmt (to delete it), this cannot be the case for a SecLabel we already have at hand. So we don't need to check if it's NULL.

```sql
┌──────────┬──────┬───────────┬──────────┬─────────┐
│  Column  │ Type │ Collation │ Nullable │ Default │
├──────────┼──────┼───────────┼──────────┼─────────┤
│ objoid   │ oid  │           │ not null │         │
│ classoid │ oid  │           │ not null │         │
│ provider │ text │ C         │ not null │         │
│ label    │ text │ C         │ not null │         │
└──────────┴──────┴───────────┴──────────┴─────────┘
```